cmd/registry/scoring/parser.go

```diff
@@ -101,36 +101,72 @@ func validateScoreFormula(targetName patterns.ResourceName, scoreFormula *rpc.Sc
 }
 
 func validateNumberThresholds(thresholds []*rpc.NumberThreshold, minValue, maxValue int32) []error {
+
+	if len(thresholds) == 0 {
+		// no error returned since thresholds are optional
+		return []error{}
+	}
+
 	errs := make([]error, 0)
 
+	// keep track of the endpoints of the ranges
+	globalRangeMax := minValue
+	globalRangeMin := maxValue
+
+	for _, t := range thresholds {
+		// Check for validity of the range values
+		rangeMin := t.GetRange().GetMin()
+		rangeMax := t.GetRange().GetMax()
+
+		// min==max is valid here, one specific value can have a dedicated severity.
+		if rangeMin > rangeMax {
+			errs = append(errs, fmt.Errorf("invalid range [%d, %d]: range.min cannot be greater than range.max", rangeMin, rangeMax))
+			continue
+		}
+
+		// Check for out of limits conditions
+		if rangeMax > maxValue || rangeMax < minValue {
+			errs = append(errs, fmt.Errorf("invalid range.max [%d, %d]: range.max(%d) should be within min_value(%d) and max_value(%d) limits", rangeMin, rangeMax, rangeMax, minValue, maxValue))
```

Comment:
Should this say "invalid range" like it does below?

Reply:
Done.